### PR TITLE
Docs about creating a service

### PR DESCRIPTION
**Pull request by davidtheclark**
Based on some conversations I've had recently, I think this doc would be helpful. 

@danswick for review, please.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -16,3 +16,48 @@
...
+
+## Creating a service
+
+First you'll create a service prototype object, then you'll pass that prototype object into the [`createServiceFactory()`](../services/service-helpers/create-service-factory.js) function.
+
+The properties of a service prototype object are the service methods. Each service method is a function that accepts a configuration object and returns a `MapiRequest`. The request should be created using `this.client.createRequest()`, which accepts an object with the following properties:
+
+- `method` (string): the HTTP method (e.g. `GET`).
+- `path` (string): the path of the endpoint, with `:express-style` colon-prefixed parameters for path parts that should be replaced by values in the `params` object.
+- `params` (object): an object whose keys correspond to the `:express-style` colon-prefixed parameters in the `path` string, and whose values are the values that should be substituted into the path. For example, with `path: '/foo/:bar/:baz'` you'll need a params object with `bar` and `baz` properties, like `{ bar: 'a', baz: 'b' }`. **You do *not* need to specify an `ownerId` param: that is automatically provided by the `MapiClient`.**
```
Comment (danswick) [inline docs/development.md]: this is really nicely explained!

File: docs/development.md [hunk 2]
```diff
@@ -16,3 +16,48 @@
...
+- `query` (object): an object that will be transformed into a query string and attached to the `path` (e.g. `{ foo: 'a', baz: 'b' }` becomes `?foo=a&baz=b`).
+- `headers` (object): an object of headers that should be added to the request. Keys should be lowercase.
```
Comment (danswick) [inline docs/development.md]: Are there any headers provided by default? Are there any that should be included in `headers` every time?

File: docs/development.md [hunk 3]
```diff
@@ -16,3 +16,48 @@
...
+- `body` (object, default `null`): a body that should be included with the `POST` or `PUT` request. It will be stringified as JSON.
```
Comment (danswick) [inline docs/development.md]: I always get tripped up stringified wording. Should this be "stringified JSON" (without the "as")? Or is "stringified as JSON" more standard?

Comment (kepta) [reply docs/development.md]: @danswick every JSON is a string in Javascript. In this case it means an Object (not a string ) -> would be stringified to become a JSON.

File: docs/development.md (hunk 2 above)
Comment (davidtheclark) [reply docs/development.md]: I'll add a note about `content-type: application/json` being automatically included if the request includes a `body`.

**Pull request merged**